lib/cosmos/dart/lib/dart_common.rb

```diff
@@ -150,7 +150,7 @@ def setup_packet_config(packet, packet_id, packet_config)
     end
 
     # Mark packet config ready
-    packet_config.max_table_index = table_index - 1
+    packet_config.max_table_index = table_index
```

Comment:
This was a tricky bug. By subtracting one you make the basic case have a max_table_index of 0. Then when you're trying to cleanup you do this: packet_config.max_table_index.times which of course doesn't run since the max_table_index is 0.